File: Guess_the_Number_Game.cpp
#include <bits/stdc++.h>


using namespace std;

int main() {
    srand(static_cast<unsigned>(time(0))); // Seed for random number generation
    int number = rand() % 100 + 1; // Random number between 1 and 100
    int guess;
    bool guessedCorrectly = false;

    cout << "Guess the number between 1 and 100: ";

    while (!guessedCorrectly) {
        cin >> guess;
        if (guess > number) {
            cout << "Too high! Try again: ";
        } else if (guess < number) {
            cout << "Too low! Try again: ";
        } else {
            cout << "Congratulations! You guessed it right." << endl;
            guessedCorrectly = true;
        }
    }

    return 0;
}

File: Guess_the_Number_Game.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function guessTheNumber() {
    const number = Math.floor(Math.random() * 100) + 1;

    function askGuess() {
        rl.question("Guess the number between 1 and 100: ", (input) => {
            const guess = parseInt(input, 10);

            if (guess > number) {
                console.log("Too high! Try again.");
                askGuess();
            } else if (guess < number) {
                console.log("Too low! Try again.");
                askGuess();
            } else {
                console.log("Congratulations! You guessed it right.");
                rl.close();
            }
        });
    }

    askGuess();
}

guessTheNumber();
